scripts/run-clang-format.sh

```diff
@@ -1,73 +1,53 @@
-#! /bin/bash
+#!/bin/bash
 #
-# run-clang-format.sh: formats source code in this repo in accordance with .clang-format file.
+# run-clang-format.sh: Formats source code in this repo in accordance with .clang-format file.
 # This file is part of the DP3 software package.
 # Copyright (C) 2020 ASTRON (Netherlands Institute for Radio Astronomy)
 # License: GNU General Public License version 3 or any later version
+#
+# To hook this script to pre-commit include the line
+# "./scripts/run-clang-format.sh" to .git/hooks/pre-commit
+# and make sure pre-commit is an executable shell script.
 
-programname=$0
+#Script configuration for this repo. Adjust it when copying to a different repo.
 
-set -e
+#The directories that contain source files, which clang-format should format.
+SOURCE_DIRS=($(dirname "$0")/..)
 
-# Function to print usage of this script
-usage () {
-    s=$(printf "%-50s" "=")
-    echo "${s// /=}"
-    echo "Script to format C++ header and source files. Expects a .clang-format file in current directory or parent"
-    echo ""
-    echo "Usage: $programname [-i filename/pattern] [-s skip directory]"
-    echo "  -i      include file(pattern), pass wild-card arguments as literals!"
-    echo "  -s      skip directory in diff. Must be specified as a path, e.g. ./external"
-    echo "  -h      display help"
-    exit 1
-}
+#Directories that must be excluded from formatting.
+EXCLUDE_DIRS=()
```

Comment:
`EXCLUDE_DIRS` now need to be specified relative to the location of the script, that's a bit inconvenient. Can be easily fixed by opting for just one `SOURCE_DIR`, and `cd` into `SOURCE_DIR` before executing `find`